core/db/db_stats_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright IBM Corp. 2016 All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		 http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package db
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric/protos"
+	"github.com/spf13/viper"
+	"github.com/tecbot/gorocksdb"
+)
+
+//This file contains a test function TestDumpDBStats. This test is not a unit test, rather this is used for printing
+// the details of a db-dump pointed to by DBDir. For analysing a db-dump,
+// - comment the t.SkipNow(),
+// - specify the location of db-dump in DBDir, and
+// - execute "go test -run TestDumpDBStats"
+
+const (
+	//MaxValueSize is used to compare the size of the key-value in db.
+	//If a key-value is more than this size, it's details are printed for further analysis.
+	MaxValueSize = 1024 * 1024
+
+	//DBDir the name of the folder that can be given for off-line study of a db dump
+	DBDir = ""
+)
+
+type detailPrinter func(data []byte)
+
+func TestDumpDBStats(t *testing.T) {
```

Comment:
@christo4ferris I was also in the similar dilemma. The intention is to make is easier to run using 'go', (one can run it as 'go test -run TestName'). 
The alternate is to have a file (say 'DumpDBStats.go') in a separate sub-folder of db (say folder 'github.com/hyperledger/fabric/core/db/tools'). The file would declare package 'main' and contain a function 'main' (so that one can run it as 'go run DumpDBStats.go'). 
I'm fine with this alternate as well. Do you prefer this?